pymatgen/io/vasp/outputs.py

```diff
@@ -658,21 +673,28 @@ def run_type(self):
             rt = "B3LYP"
         elif self.parameters.get("LHFCALC", True):
             rt = "PBEO or other Hybrid Functional"
-        elif self.parameters.get("LUSE_VDW", False):
-            if self.incar.get("METAGGA", "").strip().upper() in METAGGA_TYPES:
-                rt = METAGGA_TYPES[self.incar.get("METAGGA", "").strip().upper()] + "+rVV10"
-            else:
-                rt = GGA_TYPES[self.parameters.get("GGA", "").strip().upper()] + "+rVV10"
-        elif self.incar.get("METAGGA", "").strip().upper() in METAGGA_TYPES:
-            rt = METAGGA_TYPES[self.incar.get("METAGGA", "").strip().upper()]
-            if self.is_hubbard or self.parameters.get("LDAU", True):
-                rt += "+U"
+        elif self.incar.get("METAGGA") and self.incar.get("METAGGA") != "--":
+            incar_tag = self.incar.get("METAGGA", "").strip().upper()
+            rt = METAGGA_TYPES.get(incar_tag, incar_tag)
+        elif self.parameters.get("GGA"):
+            incar_tag = self.parameters.get("GGA", "").strip().upper()
+            rt = GGA_TYPES.get(incar_tag, incar_tag)
```

Comment:
@shyuep I think this block will already catch any new functionals VASP may implement